src/Atmos/Model/AtmosModel.jl

```diff
@@ -489,6 +489,14 @@ density(atmos::AtmosModel, state::Vars, aux::Vars) =
 density(::Compressible, state, aux) = state.ρ
 density(::Anelastic1D, state, aux) = aux.ref_state.ρ
 
+pressure(atmos::AtmosModel, state::Vars, aux::Vars) =
+    pressure(atmos.compressibility, atmos, state, aux)
+
+function pressure(::Compressible, atmos, state, aux)
+    ts = recover_thermo_state(atmos, state, aux)
```

Comment:
Based on the way that this is used, it might be better to pass `ts` as an argument.

Reply:
the anelastic1D version does not use `ts`

Reply:
Fair, but we typically know what `ts` is anyway. Right now, `recover_thermo_state` performs SA, so my suggestion is more of a performance related